server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -604,28 +676,263 @@ public boolean equals(Object o) {
                 return false;
             }
             Clusters clusters = (Clusters) o;
+            /// MP TODO: not sure what to do about successful and skipped here, since they are calculated differently
+            /// MP TODO: between CCS MRT=true and other search types
             return total == clusters.total && successful == clusters.successful && skipped == clusters.skipped;
         }
 
         @Override
         public int hashCode() {
+            /// MP TODO: not sure what to do about successful and skipped here, since they are calculated differently
+            /// MP TODO: between CCS MRT=true and other search types and are not immutable for CCS
```

Comment:
I need help here. I'm not sure what to do with these, since the Clusters object is no longer immutable under the CCS MRT=true scenario. I'm not clear when equals and hashCode is called for Clusters.

Reply:
Thanks. I removed the comments (next push).